### ISSUE TITLE
Starship prompt's directory is one command behind

### ISSUE THREAD

**Issue by avandesa** [nushell/nushell#3002]
**Describe the bug**

The current directory displayed by starship lags behind by one command when changing directories

**To Reproduce**
1. Set starship as the prompt
2. Navigate around the directory tree using `cd`
3. Observe that after two `cd`s, the displayed directory is not correct
4. Observe that after executing another command that does not change the directory (i.e., `echo` or `pwd`), the displayed directory is correct.

**Expected behavior**

The prompt should display the correct current working directory.

**Screenshots**

![image](https://user-images.githubusercontent.com/11090778/106773675-53b75b00-660f-11eb-8e29-ef2602ac3518.png)

**Configuration (please complete the following information):**

```
Documents/test/bar
❯ version | pivot
───┬────────────────────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────
 # │      Column0       │                                                 Column1
───┼────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────
 0 │ version            │ 0.26.0
 1 │ build_os           │ windows-x86_64
 2 │ rust_version       │ rustc 1.49.0 (e1884a8e3 2020-12-29)
 3 │ rust_channel       │ stable (default)
 4 │ cargo_version      │ cargo 1.49.0 (d00d64df9 2020-12-05)
 5 │ pkg_version        │ 0.26.0
 6 │ build_time         │ 2021-01-29 12:50:53
 7 │ build_rust_channel │ release
 8 │ features           │ clipboard-cli, ctrlc, default, directories, dirs, ichwh, ptree, rustyline, term, trash, uuid, which, zip
───┴────────────────────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────
```

**Add any other context about the problem here.**

* I was not able to reproduce this with either the default prompt (none set), or with `prompt = "echo $(pwd) >"
* As suggested in #2873, this may be an issue with the environment that starship looks at not being fully created.

**Comment by fdncred** [comment]
I wonder if starship is looking for a `PWD` env variable that we may not be setting. It [appears that's what happens](https://github.com/starship/starship/blob/452694903c9682b597bee5fde728a4a82795a355/src/init/starship.ps1) with starship in powershell.

And perhaps in the [code](https://github.com/starship/starship/blob/22dc8b842e721c440aa98035dd2dff30c7fe6497/src/context.rs#L52). I wonder if we can enable debug logging to see what's going on.

**Comment by mborejdo** [comment]
Unfortunately this problem also appears while using -p flag for starship.

```prompt="starship prompt -p $(pwd)"```

**Comment by mborejdo** [comment]
The fix mentioned in #2873 fixes the problem temporarily. but there seems to be a regression, where PWD seems to be set later and is not being picked up by starship ran in prompt

**Comment by gim913** [comment]
This seems to be fixed in nushell 0.30

**Comment by fdncred** [comment]
Thanks @gim913. If the issue resurfaces please re-open.
